src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.Tasks.props

```diff
@@ -18,7 +18,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Older SDKs used this property as a sentinel instead, to control the import of this file
          (but not the targets, which were included with the SDK). -->
     <UsingILLinkTasksSdk>true</UsingILLinkTasksSdk>
-    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\tools\net9.0\ILLink.Tasks.dll</ILLinkTasksAssembly>
+    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\tools\$(NetCoreAppToolCurrentVersion)\ILLink.Tasks.dll</ILLinkTasksAssembly>
```

Comment:
That one can't use a property from runtime as it is a shipping file. We need a different strategy for that file but I would keep that out of this PR. Runtime sets this to a different path so SDK upgrades aren't blocked by this hardcode.

Reply:
I think this can be a generated file from the template like https://github.com/dotnet/runtime/blob/main/eng/testing/linker/project.csproj.template (copy to artifacts/obj, followed by string replacement)